packages/spanner/test/session-pool.js

```diff
@@ -1376,23 +1345,143 @@ describe('SessionPool', function() {
       });
     });
 
-    describe('listens for next opening', function() {
-      it('should call itself again when available is emitted', function(done) {
-        sessionPool.once = function(event, onAvailable) {
-          sessionPool.once = util.noop;
+    describe('polls for next opening', function() {
+      it('should poll for the next available session', function(done) {
+        sessionPool.pollForSession_ = function(callback) {
+          callback();
+        };
+
+        sessionPool.getNextAvailableSession_(OPTIONS, done);
+      });
+    });
+  });
 
-          assert.strictEqual(event, 'available');
+  describe('pollForSession_', function() {
+    var setInterval_;
 
-          assert.strictEqual(
-            onAvailable(),
-            sessionPool.getNextAvailableSession_(OPTIONS, assert.ifError)
-          );
+    beforeEach(function() {
+      setInterval_ = global.setInterval;
+      global.setInterval = function(cb) {
+        global.setInterval.calledWith_ = arguments;
+        setImmediate(cb);
+        return 1;
```

Comment:
Can you make this something like `SET_INTERVAL_ID` for clarity when used in the later tests?